crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -23,14 +23,164 @@ pub struct Fulfillment {
     /// order.
     executed: order::TargetAmount,
     fee: Fee,
+    protocol_fee: order::SellAmount,
```

Comment:
No other reason. I liked everything in one place because if I move out protocol fee calculation, then `Fullfilment` object will have two states, one when receives raw solution from solver, other after protocol fee is applied (wherever that happens) so one needs to be aware of this when debugging and maintaining code.
But in general I am ok with both.

Reply:
deleted.